### PR TITLE
fix: Task.artifacts and Task.history null deserialization (fixes #257)

### LINKED ISSUE #257: [Bug] Task.artifacts and Task.history deserialize as null instead of empty list
### What happened?

According to the [A2A Task spec](https://agent2agent.info/docs/concepts/task/), both `artifacts` and `history` are **optional**:

```ts
interface Task {
  id: string;
  sessionId: string;
  status: TaskStatus;
  history?: Message[];    // optional
  artifacts?: Artifact[]; // optional
  metadata?: Record<string, any>;
}
```

### Current Behavior

In the Java SDK, when these fields are omitted from a Task JSON, they are deserialized as null.

#### Example:
```
{
  "kind": "task",
  "id": "task-123",
  "contextId": "ctx-123",
  "status": { "state": "completed" }
}
```

#### Deserializing this results in:
```
task.getArtifacts(); // null
task.getHistory();   // null
```

This is surprising for consumers, since in Java collections are usually modeled as empty rather than null. This forces unnecessary null checks and increases risk of NullPointerException.

### Expected Behavior

When omitted or explicitly null, these fields should default to empty lists, consistent with:
- The spec’s “optional” semantics.
- The Python SDK, which represents them as Optional[List] but defaults to [] when not present.

### Impact

- Breaks principle of least surprise for SDK users.
- Potential NPE risk when iterating over artifacts/history.
- Inconsistent with other SDKs.

### Related

- Related to #253, which reported deserialization errors. In this case, deserialization succeeds but produces null, which is a different bug.

### Suggested Fix
- Normalize artifacts and history to Collections.emptyList() when missing or null.
- Add regression tests to verify both omitted and explicitly null values result in empty lists.

### Code of Conduct

- [x] I agree to follow this project's Code of Conduct

### PR DESCRIPTION
**Pull request by Niravkumar-Patel**
# Description

- Initialize artifacts and history fields with empty lists when null
- Add comprehensive unit tests for deserialization scenarios
- Ensures type safety for optional array fields per A2A specification

Fixes #257

### CODE REVIEW

File: spec/src/main/java/io/a2a/spec/Task.java [hunk 1]
```diff
@@ -51,8 +52,8 @@ public Task(@JsonProperty("id") String id, @JsonProperty("contextId") String con
         this.id = id;
         this.contextId = contextId;
         this.status = status;
-        this.artifacts = artifacts;
-        this.history = history;
+        this.artifacts = artifacts != null ? artifacts : Collections.emptyList();
+        this.history = history != null ? history : Collections.emptyList();
```
[the remark this answers is not in the document]
Comment (Niravkumar-Patel) [reply spec/src/main/java/io/a2a/spec/Task.java]: Updated.

**Review by kabir — APPROVED** [review]
